mariadb_kernel/kernel.py

```diff
@@ -141,10 +143,52 @@ def do_execute(
 
         return rv
 
-    def do_shutdown(self, restart):
-        self.mariadb_client.stop()
+    def num_connected_clients(self):
+        sql_query = "select count(id) from information_schema.processlist;"
```

Comment:
I've tested the patch locally and it doesn't work as expected. The last kernel alive won't kill the server because information_schema.processlist also includes some system users internal to MariaDB server, and the count is always bigger than 1.
The query we should use that would work for us is: `show status like 'Threads_connected';`. This will return the real number of clients connected to the server (it does have its downsides though when the server is configured to use a thread pool, but because we want to check if the count is lower than 1, that downside doesn't affect us).

Reply:
Strange, it always work for me. In my testing I install mariadb from Ubuntu libraries, maybe a difference?

Well,  I am fine with any query that works, I just happened to use the first Google showed me).  will change :-)

Reply:
You're most probably right, those system processes are included in 10.5+ and probably you got the version before that, ups, sorry 😆 I think Threads_connected should be more stable across versions.